jbmc/src/java_bytecode/ci_lazy_methods_needed.cpp

```diff
@@ -112,7 +112,7 @@ void ci_lazy_methods_neededt::gather_field_types(
   {
     // If class_type is not a symbol this may be a reference array,
     // but we can't tell what type.
-    if(class_type.id() == ID_symbol)
+    if(class_type.id() == ID_symbol_type)
```

Comment:
To avoid this pain in the future, in the TG bump I am going to use `can_cast_type<symbol_typet>(class_type)` rather than directly accessing the ID - you might like to do something similar with this PR while the TG bump is being prepared :blush:

Reply:
This should be one-off -- we should simply stick to the convention that the id() matches the name of the class type.

Reply:
> should be considered to rename it all to typedef_type
Seems like it could change at least one more time. 

In any case, this pattern, were it to be used more widely, would hide more of the internal ireps and make changes like this less painful in future. 

It also allows for at a later date adding more rigorous type checking (e.g. a `symbol_typet` should always have one named sub, `ID_identifier` etc).